### DISCUSSION TITLE
Updated to latest version old Gas gone but new error logs

### DISCUSSION THREAD

**Discussion by thorrrr** [BottlecapDave/HomeAssistant-OctopusEnergy#352] [General]
Updated to latest version old Gas gone but new error logs  do i need to add a bug or is it a setting i need to change?

Logger: homeassistant.components.recorder.db_schema
Source: components/recorder/db_schema.py:584
Integration: Recorder (documentation, issues)
First occurred: 07:16:24 (1125 occurrences)
Last logged: 08:47:48

State attributes for sensor.octopus_energy_electricity_redacted_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
State attributes for sensor.octopus_energy_gas_redacted_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
State attributes for camera.xiaomi_cloud_map_extractor exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored

Other Log

Logger: homeassistant.components.sensor.recorder
Source: components/sensor/recorder.py:405
Integration: Sensor (documentation, issues)
First occurred: 07:20:10 (2 occurrences)
Last logged: 07:20:10

The unit of sensor.octopus_energy_electricity_redacted_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
The unit of sensor.octopus_energy_gas_redacted_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this

I am also seeing the previous days figures have stopped showing after update ! (see bottom images) this was working

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/12d0b775-7968-43d9-8bf8-952d4422699f)

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/c3be5878-39b3-4437-9000-28951f7b8602)

![image](https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/assets/8170673/5623c071-5536-45b9-939f-e5a252d3b2a0)

**Comment by BottlecapDave** [comment]
Hi there. 

> Logger: homeassistant.components.recorder.db_schema
Source: components/recorder/db_schema.py:584
Integration: Recorder (documentation, issues)
First occurred: 07:16:24 (1125 occurrences)
Last logged: 08:47:48
State attributes for sensor.octopus_energy_electricity_redacted_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
State attributes for sensor.octopus_energy_gas_redacted_current_rate exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
State attributes for camera.xiaomi_cloud_map_extractor exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored

These are warnings not errors, probably caused by the addition of data that we're now looking at in terms of rate information that pushed it over the edge. As you can see, other integrations (xiaomi) also experience this. Based on the message, I don't think it's anything to worry about and just means the attributes of the sensor wont be saved to the database for context. 

The only way of removing this warning is not having the rates or applicable rates available on the entity, but that seems counterintuitive as this is useful for some (especially when debugging). Another option is to expose the data as a new entity, but there is no entity that fits the shape of data. There is also no way of telling HA to have attributes that are exposed but not saved.

>Logger: homeassistant.components.sensor.recorder
Source: components/sensor/recorder.py:405
Integration: Sensor (documentation, issues)
First occurred: 07:20:10 (2 occurrences)
Last logged: 07:20:10
The unit of sensor.octopus_energy_electricity_redacted_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this
The unit of sensor.octopus_energy_gas_redacted_current_rate (None) cannot be converted to the unit of previously compiled statistics (GBP/kWh). Generation of long term statistics will be suppressed unless the unit changes back to GBP/kWh or a compatible unit. Go to https://my.home-assistant.io/redirect/developer_statistics to fix this

This is another warning which I believe is because the data is not available yet. Based on the time of the first occurred log, it looks like you restarted the integration (or your HA instance) this morning when the previous consumption data isn't available (usually around midday for electricity and abit later for gas but can vary per account). It should become available by the end of the day. Nothing was changed in the latest release in these areas. If it's not available by the end of the day, then raise an issue.
